### PR TITLE
Yia - weather app - design 2

### PR DESCRIPTION
**Pull request by Citronskal**
## Netlify link
https://yias-weather-app.netlify.app/ 

## Collaborators
Add your collaborators here. Write their GitHub usernames in square brackets. If there's more than one, separate them with a comma, like this:
[none]

### CODE REVIEW

**Comment by Yifan-858** [comment]
Hey Yia, great work! Your code has a well-organized structure. That makes it very easy to follow along! The app loads quickly and the weather data is retrieved efficiently. Overall, your project meets the requirements. Keep going!

File: code/Design 2/index.js [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+
+// Event listener for weather form submission
+weatherForm.addEventListener("submit", async (event) => {
+  event.preventDefault();
+  const city = cityInput.value;
+
+  if (city) {
+    try {
+      // Fetch current weather and forecast data
+      const { currentWeather, forecast } = await getWeather(city);
```
Comment (Yifan-858) [inline code/Design 2/index.js:20]: Super efficient to use async/await to avoid callback hell!

File: code/Design 2/index.js [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+    } else {
+      cityInput.style.display = "none";
+    }
+  }
+});
+
+//Fetch weather data
+async function getWeather(city) {
+  const currentWeatherUrl = `https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${apiKey}`;
+  const forecastUrl = `https://api.openweathermap.org/data/2.5/forecast?q=${city}&appid=${apiKey}`;
```
Comment (Yifan-858) [inline code/Design 2/index.js:67]: "&units=metric" can be added to the URL("api.openweathermap.org/data/2.5/weather?q=${city}**_&units=metric_**&appid=${apiKey}") to get Celsius temperature data as mentioned in Open weather API doc. That will save you some lines for converting the units of the temp :)

File: code/Design 2/index.js [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+
+  // Apply different background based on whether it's night or day
+  if (isNight) {
+    body.style.background = "linear-gradient(to right, #333333, #000000)";
+    body.style.color = "#ffffff";
+    iconElement.src = `assets/moon.png`;
+  } else {
+    body.style.background = "linear-gradient(#8589ff, #e8e9ff)";
+    body.style.color = "#ffffff";
+  }
```
Comment (Yifan-858) [inline code/Design 2/index.js:134]: Being a bit nitpicky here, I will keep the styling parts in the CSS file to make the codes easier to maintain and read for the other developers in the future. :)

File: code/Design 2/index.js [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+    console.error(error);
+    displayError("Failed to fetch weather forecast. Please try again later.");
+  }
+}
+
+function displayError(message) {
+  const errorDisplay = document.createElement("p");
+  errorDisplay.textContent = message;
+  errorDisplay.classList.add("error");
+  weatherBackground.appendChild(errorDisplay);
```
Comment (Yifan-858) [inline code/Design 2/index.js:279]: Good idea to add an error message!

File: code/Design 2/style.css [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+  flex-wrap: wrap;
+  align-items: center;
+  justify-content: center; /* Center the items horizontally */
+  font-weight: 300;
+  width: 100%;
+  max-width: 800px;
+  margin: 0 auto; /* Center the section horizontally */
+  padding: 20px; /* Add padding for spacing */
+  margin-top: 20px; /* Add top margin */
+  margin-bottom: 20px; /* Add bottom margin */
```
Comment (Yifan-858) [inline code/Design 2/style.css:161]: margin can be combined like this: 
`margin: 20px auto;`

**Review by AntonellaMorittu — CHANGES_REQUESTED** [review]
Hi Yia, very good job this week! Your code is well structured and formatted and the app is working well, you completed the basic requirements + search bar⭐ 
There are a couple of things to fix to meet the requirements:

- The app should have: 4-day forecast You are displaying all 5 days when we requested 4 
- the temperature should be rounded to 1 decimal
- the UI needs to follow the design as close as possible

Not a big deal. Fix this and you're good to go:) Keep up the awesome work 🚀

**Review by AntonellaMorittu — CHANGES_REQUESTED** [review]
Nice fixing the decimals, now last thing to fix is the layout for mobile, it's offset (see screenshot) please fix this, almost there 💪 
Screenshot 2024-03-08 at 13 30 22

**Comment by HIPPIEKICK** [comment]
Still the same. Feel free to reach out on SO if you need help/inspiration 😊 

Skärmavbild 2024-03-12 kl  14 43 09

**Comment by Citronskal** [comment]
@HIPPIEKICK I fixed the issue I think, can you take a look for me? Thank you!

**Review by HIPPIEKICK — APPROVED** [review]

**Pull request closed without merging**